src/lib.rs

```diff
@@ -63,6 +102,17 @@ pub struct PmsData {
     checksum: u16,
 }
 
+// Calculates Air Quality Index (AQI) 
+// Formula: https://en.wikipedia.org/wiki/Air_quality_index#Computing_the_AQI
+fn calculate_aqi(breakpoints: &[(f64, f64)], data: f64) -> f64 {
+    if data <= 0.0 { return 0.0; }
+    let data_nearest_tenth = (data * 10.0).round() / 10.0;
```

Comment:
This is effectively a fixed point floating point number, which may be better to store as an integer since floating points have some rounding errors (then the breakpoints could be integer tenths too, e.g., 120 instead of 12.0). That said, I am not sure I understand the point of rounding to the nearest tenth. Could you explain?

Reply:
Good question.  The intent is to keep the breakpoint values 100% in sync with the [published values](https://aqs.epa.gov/aqsweb/documents/codetables/aqi_breakpoints.html).  Rounding to the nearest tenth helps align the sensor f64 data to the breakpoint boundaries.  Happy to add comments if you think that would help explain.